core/src/test/java/io/questdb/griffin/SqlCompilerTest.java

```diff
@@ -2546,6 +2546,40 @@ public void testInsertAsSelectColumnSubset2() throws Exception {
         );
     }
 
+    @Test
+    public void testInsertAsSelectNull() throws Exception {
+        testInsertAsSelect("a\tb\n" +
+                        "\uFFFF\t\n" +
+                        "\uFFFF\t\n" +
+                        "\uFFFF\t\n",
+                "create table x (a CHAR, b SYMBOL)",
+                "insert into x " +
+                        "select" +
+                        " null," +
+                        " null" +
+                        " from long_sequence(3)",
+                "x");
+    }
+
+    @Test
+    public void testInsertAsSelectNullInconvertible() throws Exception {
+        try {
+            testInsertAsSelect("nothing",
+                    // in this context null is the default type, STRING
+                    "create table x (a CHAR, b LONG)",
+                    "insert into x " +
+                            "select" +
+                            " null," +
+                            " null" +
+                            " from long_sequence(3)",
+                    "x");
+            Assert.fail();
```

Comment:
This should not fail, null should not be treated as a STRING but as a LONG here, ideally... Or put a comment that it's not done yet